eng/pipelines/templates/steps/vcpkg.yml

```diff
@@ -0,0 +1,48 @@
+# Assumes that the VM image is in a variable called OSVmImage
```

Comment:
Would be better to assert that variables['OSVmImage'] is set

Reply:
I like the idea of templates doing runtime validation. I'll chat with @weshaggard about what this might look like in a broader sense across all of our engineering systems.

Reply:
What we have done in other places is to setup a template parameter and default the value to the variable we want to use. While that isn't exactly an assert it at least adds some documentation as well as enables folks to override the default if needed.